lib/ocpp/v201/charge_point.cpp

```diff
@@ -631,6 +605,169 @@ void ChargePoint::configure_message_logging_format(const std::string& message_lo
     }
 }
 
+void ChargePoint::handle_cost_and_tariff(const TransactionEventResponse& response,
+                                         const TransactionEventRequest& original_message,
+                                         const json& original_transaction_event_response) {
+    const bool tariff_enabled = this->is_tariff_enabled();
+
+    const bool cost_enabled = this->is_cost_enabled();
+
+    std::vector<DisplayMessageContent> cost_messages;
+
+    // Check if there is a tariff message and if 'Tariff' is available and enabled
+    if (response.updatedPersonalMessage.has_value() && tariff_enabled) {
+        MessageContent personal_message = response.updatedPersonalMessage.value();
+        DisplayMessageContent message = message_content_to_display_message_content(personal_message);
+        cost_messages.push_back(message);
+
+        // If cost is enabled, the message will be sent to the running cost callback. But if it is not enabled, the
+        // tariff message will be sent using the display message callback.
+        if (!cost_enabled && this->callbacks.set_display_message_callback.has_value() &&
+            this->callbacks.set_display_message_callback != nullptr) {
+            DisplayMessage display_message;
+            display_message.message = message;
+            display_message.identifier_id = original_message.transactionInfo.transactionId;
+            display_message.identifier_type = IdentifierType::TransactionId;
+            this->callbacks.set_display_message_callback.value()({display_message});
+        }
+    }
+
+    // Check if cost is available and enabled, and if there is a totalcost message.
+    if (cost_enabled && response.totalCost.has_value() && this->callbacks.set_running_cost_callback.has_value()) {
```

Comment:
small nitpick: we typically use "and" "or" "not" instead of "&&" "||" "!"

Reply:
ok done for all files i touched.